### PR TITLE
make affinity configurable for Reaper

### LINKED ISSUE #847: Affinity configuration support for Reaper pod scheduling
**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
Provide support within the k8ssandra chart for the specification of affinity configuration to allow for further control of the scheduling of reaper pods within a deployment.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-502) by [Unito](https://www.unito.io)
┆Epic: Pod Scheduling Control
┆Fix Versions: k8ssandra-1.3.0

### PR DESCRIPTION
**Pull request by jsanda**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #847 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/k8ssandra/values.yaml [hunk 1]
```diff
@@ -455,8 +455,12 @@ reaper:
     traefik:
       # -- Traefik entrypoint where traffic is sourced. See https://doc.traefik.io/traefik/routing/entrypoints/
       entrypoint: web
-      # -- Tolerations to apply to the Reaper pods. See
-      # https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/ for background.
+  # -- Affinity to apply to the Stargate pods. See
```
Comment (jdonenine) [inline charts/k8ssandra/values.yaml:458]: Very minor, but this should say Reaper not Stargate.

File: charts/reaper-operator/templates/role.yaml [hunk 2]
```diff
@@ -72,4 +69,61 @@ rules:
...
       - get
       - patch
-      - update
\ No newline at end of file
+      - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: leader-election-role
```
Comment (jdonenine) [inline charts/reaper-operator/templates/role.yaml:77]: What are these leader election roles used for?

**Review by jdonenine — COMMENTED** [review]
All of my testing went well, was able to manipulate the deployment with the new settings, very nice!

Just a minor typo and a question about the role that seems out of place.

Thanks!

File: charts/k8ssandra/values.yaml (hunk 1 above)
Comment (jsanda) [reply charts/k8ssandra/values.yaml:458]: Nice catch. I'll fix that.

File: charts/reaper-operator/templates/role.yaml (hunk 2 above)
Comment (jsanda) [reply charts/reaper-operator/templates/role.yaml:77]: controller-runtime, the main library on which operators/controllers are built, provides leader election. The leader election stuff is generated with scaffolding.

Comment (jdonenine) [reply charts/reaper-operator/templates/role.yaml:77]: 👍  thanks

**Review by jdonenine — APPROVED** [review]

**Pull request merged**